daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -141,11 +141,12 @@ class Engine(val config: EngineConfig = new EngineConfig(LanguageVersion.StableV
     for {
       commandWithCids <- preprocessor.preprocessCommand(command)
       (speedyCommand, globalCids) = commandWithCids
+      sexpr = compiledPackages.compiler.unsafeCompileForReinterpretation(speedyCommand)
```

Comment:
Maybe we can extend the comment to explain what's happening when the command fails with an unhanded exception ?

Reply:
done